### PR TITLE
Event pagination doc

### PR DESCRIPTION
**Pull request by patrickkuo**
Added docs for event query and examples

### CODE REVIEW

**Review by randall-Mysten — COMMENTED** [review]
Few comments. I can make these changes later as well. Note that deleting pubsub also requires a change the the navconfig
https://github.com/MystenLabs/sui/blob/main/doc/src/navconfig.json#L200

File: doc/src/build/event_api.md [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+{
+  "checkpoint": 10
+}
+```
+
+# Sui event query
+
+## Event query criteria
+
+Sui event query api uses `EventQuery` criteria object to enable ;
```
[the remark this answers is not in the document]
Comment (patrickkuo) [reply doc/src/build/event_api.md:137]: EventQuery is an query criterion object user provide when calling the Event query api

[the remark this answers is not in the document]
Comment (randall-Mysten) [reply doc/src/build/event_api.md:137]: sorry I was not clear. If it is an API it should be `EventQuery` API. If it is a code element it is helpful to format as code
I am asking about the lower case "Sui event query API uses..."
Is it the proper Event Query API? If referring conceptually to it, it is a proper noun and should be Event 
But isn't Event Query a method the Sui API not an API itself?

**Comment by randall-Mysten** [comment]
The deletion of pubsub will result in a broken link in the nav. Could you also remove that. I can get it in a follow up PR if not.
https://github.com/MystenLabs/sui/blob/main/doc/src/navconfig.json#L198

**Comment by patrickkuo** [comment]
> The deletion of pubsub will result in a broken link in the nav. Could you also remove that. I can get it in a follow up PR if not.
> https://github.com/MystenLabs/sui/blob/main/doc/src/navconfig.json#L198

Thanks! @randall-Mysten !!

Please feel free to put in any edit, also please let me know if there are any content missing or anything more I should add.

**Comment by randall-Mysten** [comment]
> > The deletion of pubsub will result in a broken link in the nav. Could you also remove that. I can get it in a follow up PR if not.
> > https://github.com/MystenLabs/sui/blob/main/doc/src/navconfig.json#L198
> 
> Thanks! @randall-Mysten !!
> 
> Please feel free to put in any edit, also please let me know if there are any content missing or anything more I should add.

LGTM. Thanks for updating docs. I may make some additional copyedits after this is published, but nothing else for this PR.

**Review by randall-Mysten — APPROVED** [review]

**Pull request merged**